libs/langchain/langchain/chat_models/tongyi.py

```diff
@@ -318,7 +318,14 @@ def _generate(
             )
             return _generate_from_stream(stream_iter)
 
+        if not messages:
+            raise ValueError("No messages provided.")
+        
         message_dicts, params = self._create_message_dicts(messages, stop)
+
+        if message_dicts[-1]['role'] != 'user':
```

Comment:
Why is this restriction needed? Does the API error when the last message is not `user` or does it return something?

Reply:
Yes, if the last message is not `user`, the API will return this:
![image](https://github.com/langchain-ai/langchain/assets/45618243/077881f0-c06f-463b-94d0-677f0565319a)
which doesn't give an explicit reason.